libcaf_core/caf/flow/for_each.test.cpp

```diff
@@ -58,29 +46,31 @@ SCENARIO("for_each iterates all values in a stream") {
         /* subtest */ {
           auto inputs = std::vector<int>{1, 2, 4, 8, 16, 32, 64, 128};
           auto outputs = std::vector<int>{};
-          ctx->make_observable()
+          make_observable()
             .from_container(inputs) //
             .as_observable()
             .filter([](int) { return true; })
             .for_each([&outputs](int x) { outputs.emplace_back(x); });
-          ctx->run();
-          CHECK_EQ(inputs, outputs);
+          run_flows();
+          check_eq(inputs, outputs);
         }
         /* subtest */ {
           auto completed = false;
           auto inputs = std::vector<int>{21, 21, 21, 21, 21, 21, 21};
           auto outputs = std::vector<int>{};
-          ctx->make_observable()
+          make_observable()
             .repeat(7) //
             .as_observable()
             .take(7)
             .map([](int x) { return x * 3; })
-            .do_on_error([](const error& err) { FAIL("on_error: " << err); })
+            .do_on_error([this](const error& err) {
+              fail(std::string_view("on_error: " + caf::to_string(err)));
```

Comment:
```suggestion
              fail("on_error: " + caf::to_string(err));
```

Reply:
@Neverlord We can not use `fail("on_error: " + caf::to_string(err));` as we will face the following error `cannot convert from 'std::basic_string<char,std::char_traits<char>,std::allocator<char>>' to 'caf::format_string_with_location'`.

Reply:
Right, this takes a format string. Then there's no need at all to build a string in the first place: `fail("on_error: {}", caf::to_string(err))`. 🙂